apple/OmnivoreKit/Sources/App/Views/WebReader/WebReader.swift

```diff
@@ -84,11 +85,20 @@ struct WebReader: PlatformViewRepresentable {
   private func updatePlatformView(_ webView: WKWebView, context: Context) {
     if annotationSaveTransactionID != context.coordinator.lastSavedAnnotationID {
       context.coordinator.lastSavedAnnotationID = annotationSaveTransactionID
-      (webView as? OmnivoreWebView)?.dispatchEvent(.saveAnnotation(annotation: annotation))
+      do {
```

Comment:
@satindar this is the main change here, we let the WebReader control either the modal is closed and only close it if saving succeeds.